### PR TITLE
[SpellCheck] Add cspell check workflow

### PR DESCRIPTION
**Pull request by 0mza987**
# Description

- Add cspell check workflow
- Fix all existing spell errors..............

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/workflows/spell_check.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Spell check CI
```
Comment (wangchao1230) [inline .github/workflows/spell_check.yml:1]: Looking forward to the completion of this PR!

Comment (wangchao1230) [reply .github/workflows/spell_check.yml:1]: @eniac871 FYI

Comment (0mza987) [reply .github/workflows/spell_check.yml:1]: I'm getting a headache on doing this

**Review by crazygao — APPROVED** [review]

File: .cspell.json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    "pfutil",
+    "mgmt",
+    "wsid",
+    "westus",
+    "msrest",
+    "cref",
+    "msal"
+  ],
+  "ignoreWords": [
+    "openmpi",
```
Comment (wangchao1230) [inline .cspell.json:75]: what's the difference between words and ignoreWords?

Comment (0mza987) [reply .cspell.json:75]: Funtionally they are not so different, but `words` tends to mean domain specific word and ignore words are ... emm, other words we don't want spell to raise error.

**Review by wangchao1230 — APPROVED** [review]

File: examples/flows/evaluation/eval-perceived-intelligence/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# Percieved Intelligence Evaluation
+# perceived Intelligence Evaluation
```
[the remark this answers is not in the document]
Comment (0mza987) [reply examples/flows/evaluation/eval-perceived-intelligence/README.md:1]: Good catch, I'll fix in the next pr

**Pull request merged**